cheroot/test/test_cli.py

```diff
@@ -0,0 +1,81 @@
+"""Tests to verify the command line interface."""
+# -*- coding: utf-8 -*-
+# vim: set fileencoding=utf-8 :
+import sys
+
+import six
+import pytest
+
+from cheroot.cli import (
+    Application,
+    parse_wsgi_bind_addr,
+)
+
+
+@pytest.mark.parametrize(
+    'raw_bind_addr, expected_bind_addr', (
+        ('192.168.1.1:80', ('192.168.1.1', 80)),
+        ('[::1]:8000', ('::1', 8000)),
+    ),
+)
+def test_parse_wsgi_bind_addr_for_tcpip(raw_bind_addr, expected_bind_addr):
+    """Check the parsing of the --bind option for TCP/IP addresses."""
+    assert parse_wsgi_bind_addr(raw_bind_addr) == expected_bind_addr
+
+
+def test_parse_wsgi_bind_addr_for_unix_socket():
+    """Check the parsing of the --bind option for UNIX Sockets."""
+    assert parse_wsgi_bind_addr('/tmp/cheroot.sock') == '/tmp/cheroot.sock'
+
+
+def test_parse_wsgi_bind_addr_for_abstract_unix_socket():
+    """Check the parsing of the --bind option for Abstract UNIX Sockets."""
+    assert parse_wsgi_bind_addr('@cheroot') == '\0cheroot'
+
+
+class WSGIAppMock:
+    """Mock of a wsgi module."""
+
+    def application(self):
+        """Empty application method.
+
+        Default method to be called when no specific callable
+        is defined in the wsgi application identifier.
+
+        It has an empty body because we are expecting to verify that
+        the same method is return no the actual execution of it.
+        """
+
+    def main(self):
+        """Empty custom method (callable) inside the mocked WSGI app.
+
+        It has an empty body because we are expecting to verify that
+        the same method is return no the actual execution of it.
+        """
+
+
+@pytest.mark.parametrize(
+    'wsgi_app_spec, pkg_name, app_method, mocked_app', (
+        ('mypkg.wsgi', 'mypkg.wsgi', 'application', WSGIAppMock()),
```

Comment:
`WSGIAppMock()` is evaluated during the test module import time. Better use a fixture.

Reply:
The mock instance is directly related to the other params. I did liked the idea of passing it along the others params to express the direct relationship.

For example a future param `('a.b:foo', 'a.b', 'foo')` will break the test, because the mock doesn't define a `foo` method. This is not obvious via the implicit mock injection of pytest.

But I understand if you have a strong opinion on going over the pytest mock injection route.

Reply:
In this case, you'd have to deal with this on the fixture level. This is a pytest way.